controller/src/controller.rs

```diff
@@ -261,10 +325,32 @@ fn sort_shadows(shadows: &mut Vec<BottlerocketShadow>, associated_brs_name: &str
     }
 }
 
+#[instrument(skip(nodes))]
+fn find_unlabeled_nodes(mut nodes: Vec<Node>) -> Vec<String> {
+    let mut unlabeled_nodes: Vec<String> = Vec::new();
+    for node in nodes.drain(..) {
+        if !check_brupop_label(&node.clone()) {
+            unlabeled_nodes.push(node.name());
+        }
+    }
+
+    unlabeled_nodes
+}
+
+#[instrument(skip(node))]
+fn check_brupop_label(node: &Node) -> bool {
```

Comment:
A more descriptive name for the function should be better :+1:, something like ` node_has_label` or `is_node_labeled_with_<your_label_name_here>`